Eastern-Research-Group/mywaterway
# Adding code coverage support to the normal code base

## Main Changes:
* Added new coverage npm command in client
* Added new coverage npm command in /app/package.json
* Added instrumenting plugin to react client - without needing to eject (https://github.com/cypress-io/instrument-cra)
* adding other instrumenting packages listed in (https://docs.cypress.io/guides/tooling/code-coverage.html)

## Steps To Test:
1. Grab this branch
2. navigate to the /app folder 
3. run npm run coverage
4. this will kick off the instrumented react code command and open cypress.io
5. run the cypress.io tests
6. once done running, you should see two new folders in the /app folder
7. one folder called ".nyc_output" contains the .nyc specific output that can be fed into other programs like SonarQube
8. the other folder labelled "coverage" has and html formatted version of the coverage information as well as the lcov.info file which can also be injected into SonarQube to provide coverage information metrics.
9. Neither of the two new folders should be checked into version control (as such, those two folders are listed in the gitignore file)


## TODO:
- Integrate the report creation create process as part of our DevOps workflow.
